agent/testagent.go

```diff
@@ -185,7 +189,12 @@ func (a *TestAgent) Start(t *testing.T) error {
 		}
 		result, err := config.Load(opts)
 		if result.RuntimeConfig != nil {
-			result.RuntimeConfig.Telemetry.Disable = true
+			// If "telemetry" settings are present, then we probably want
+			// to actually emit consul specific metrics
+			// otherwise, disable telemetry
+			if !strings.Contains(a.HCL, "telemetry") {
+				result.RuntimeConfig.Telemetry.Disable = true
```

Comment:
It's unfortunate our test helper makes this so difficult. I wonder if we can test this without using `TestAgent`. In general I find things like `TestAgent` cause problems. Ideally we would make the real production interfaces nice enough so that we don't require complex test helpers to test them.

We have a couple tests that do not use `TestAgent`. I can help investigate that option.